MarkBind/markbind
# Hoist `<script>` and `<style>` tags to after `<body>` tag

**What is the purpose of this pull request?**




This addresses the seventh item in [#1534 (comment)](https://github.com/MarkBind/markbind/pull/1534#issuecomment-827530954).

**Overview of changes:**
- Removed unescaping of special characters after SSR
- Combined processing of script and style tags
- Hoist script and style tags found during the processing of page / layout to the bottom of the page

**Anything you'd like to highlight / discuss:**
If there's a better way to implement this and if I missed out anything.

**Testing instructions:**
`npm run test`

Create a new MarkBind project and try adding `<style>` and/or `<script>` tags in either the layout page or a normal page. 

You should find that regardless of layout page or a normal page, all `<style>` and `<script>` will be hoisted to the bottom of the rendered HTML file (when you view the source tab in developer console). 



**Proposed commit message: (wrap lines at 72 characters)**
Hoist `<script>` and `<style>` tags to after `<body>` tag.

Our setup compiles whatever is in the content section of the page into
Vue application. We also do server-side rendering on the content. Within
the content section, we allow users to write `<script>` and/or `<style>` 
tags. However, during compilation, Vue will ignore and discard these tags.
Furthermore, during server-side rendering, special characters in 
`<script>` tags will be escaped. While we have measures to tackle the 
problems above, we should look to implement a better workaround.

Let's remove the temporary measures and prevent the compilation and
rendering of `<script>` and/or `<style>`, by hoisting them out of the 
content section and to after `<body>` tag.



---

**Checklist:** :ballot_box_with_check:
